WordPress/gutenberg
# Editable: Rename the Editable component, RichText component for clarity

refs #4776

This PR renames the `Editable` component `RichText` to clarify that it's meant to be used for Rich Content exclusively and not as a replacement to inputs/textareas.

We still export an `Editable` component for backwards compatibility with a warning asking people to use `RichText` instead.

Next step will be to implement a `PlainText` component and use it in one/two blocks. The blocks where we're currently using `TextareaAutosize`.

**Testing instructions**

 - Open the editor
 - Insert different content blocks (paragraph, list, quotes, image with captions...)
 - Check that they still work and look as expected.
